gammapy/datasets/map.py

```diff
@@ -539,15 +549,16 @@ def excess(self):
     def models(self, models):
         """Models setter."""
         self._evaluators = {}
-
         if models is not None:
             models = DatasetModels(models)
             models = models.select(datasets_names=self.name)
-
+            if models:
+                psf = self._psf_kernel
```

Comment:
what happens if the psf map does not contain a one bin PSFMap?

Reply:
`self._psf_kernel ` return `None` which is the default of the `MapEvaluator`